### PR TITLE
resolve #105: add test dir and init system testing env

### LINKED ISSUE #105: add cita-test-chain for local development and CI testing

### PR DESCRIPTION
**Pull request by rainchen**
# Testing

### First of all, prepare cita-node docker image

build base image for cita-node: `$ make build CITA_VERSION=0.24.0`

### To start 1 node

1. generate node config: `$ make config-node`
2. start 1 node: `$ make start-node`
3. query block number: `$ make rpc-block-number`

### To start 4 nodes

1. generate nodes config: `$ make config-nodes`
2. start 4 node: `$ make start-nodes`
3. query block number: `$ make rpc-block-number`

### To start any number of node

1. generate nodes config: `$ make config-nodes NODES_CONFIG=node0:4000,node1:4000,node2:4000,node3:4000,node4:4000`
2. start 5 nodes: `$ make start-nodes NODES="node0 node1 node2 node3 node4"`
3. watch block number: `$ make watch TARGET=rpc-block-number`
4. stop 1 node: `$ make stop-nodes NODES="node3"`
5. watch step 3, block number should be still growing

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	@jq --version || brew install jq || apt-get install jq || (echo "install jq: https://stedolan.github.io/jq/download/" && exit 1)
+
+##@ Cleanup
+clean: ## Clean up.
+	$(info Cleaning up things, remove all containers/network:)
+	docker-compose down
+
+##@ Building
+
+build: clean deps ## Building the base docker image, e.g.: $ make build CITA_VERSION=0.24.0.
```
Comment (blankwu) [inline test/Makefile:43]: 这是先操作 clean ，然后再 build？

File: test/Makefile [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	$(info Building the base docker image)
+	docker-compose build --build-arg CITA_VERSION=$${CITA_VERSION:=0.24.0} cita_node
```
Comment (blankwu) [inline test/Makefile:45]: 两个 PR 都修改了这个内容，关闭？

**Review by blankwu — APPROVED** [review]

File: test/Makefile (hunk 1 above)
Comment (rainchen) [reply test/Makefile:43]: 这是设置2个前置步骤，意思是执行 make build 之前，先执行 make clean 再执行 make deps，如果2步都有问题，不会执行make build

File: test/Makefile (hunk 2 above)
Comment (rainchen) [reply test/Makefile:45]: 哪2个PR？ 这个是 test 目录下的 Makefile

**Pull request merged**